### PR TITLE
test(controller): add integration tests for renderedrelease core functions

### PR DESCRIPTION
**Pull request by VajiraPrabuddhaka**
## Purpose

The `renderedrelease` controller lacked integration test coverage for its core reconciliation functions (`ensureNamespaces`, `applyResources`, `deleteResources`, `listLiveResourcesByGVKs`). These functions handle namespace provisioning, server-side apply of workload resources, stale resource cleanup, and live resource discovery. All these critical paths have zero prior coverage.

## Approach

- Add integration tests for `ensureNamespaces`: empty list no-op, namespace creation, idempotency when namespace already exists, and multiple namespaces
- Add integration tests for `applyResources`: empty list no-op, server-side apply with tracking labels verified, idempotent re-apply
- Add integration tests for `deleteResources`: empty list no-op, deletion of a previously applied resource verified via NotFound
- Add integration tests for `listLiveResourcesByGVKs`: no matching resources returns empty, label-matched resources found, unlabelled resources excluded, unknown GVK continues without error, multiple GVKs queried and results collected
- All tests use `envtest` with `k8sClient` as the plane client, following the existing patterns in the package

## Related Issues

N/A

## Checklist
- [x] Tests added or updated (unit, integration, etc.)
- [ ] Samples updated (if applicable)
- [ ] Added `backport/<release-branch>` label if this should be backported (e.g., `backport/release-v1.0`)

### CODE REVIEW

File: internal/controller/renderedrelease/controller_integration_test.go [hunk 1]
```diff
@@ -307,3 +311,295 @@ var _ = Describe("RenderedRelease Controller", func() {
...
+
+	Context("when an unknown GVK is queried", func() {
+		It("should continue without returning an error", func() {
+			r := &Reconciler{Client: k8sClient, Scheme: k8sClient.Scheme()}
+			unknownGVK := schema.GroupVersionKind{Group: "unknown.example.com", Version: "v1", Kind: "NonExistentResource"}
+			release := makeRelease("some-uid-unknown-gvk")
+			result, err := r.listLiveResourcesByGVKs(ctx, k8sClient, release, []schema.GroupVersionKind{unknownGVK})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(result).To(BeEmpty())
+		})
```
[the remark this answers is not in the document]
Comment (VajiraPrabuddhaka) [reply internal/controller/renderedrelease/controller_integration_test.go:582]: I have addressed this. can you please check again?
